### PR TITLE
Update terraform-validator to v0.4.0

### PR DESCRIPTION
**Pull request by vovinacci**
Closes https://github.com/terraform-google-modules/terraform-example-foundation/issues/418.

This PR updates `terraform-validator` version and mentions to `v0.4.0`, fixes various minor documentation issues and updates `cloudbuild_bootstrap` module to `2.3`.

### CODE REVIEW

File: 0-bootstrap/README.md [hunk 1]
```diff
@@ -113,7 +108,7 @@ your current Jenkins manager (master) environment.
     ```
 1. Run `terraform init`.
 1. Run `terraform plan` and review the output.
-1. To run terraform-validator steps please follow the [instructions](https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#install-terraform-validator) in the **Install Terraform Validator** section and install version `2021-03-22`. You will also need to rename the binary from `terraform-validator-<your-platform>` to `terraform-validator`and the terraform-validator binary must be in your PATH.
+1. To run terraform-validator steps please follow the [instructions](https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#install-terraform-validator) in the **Install Terraform Validator** section and install version `v0.4.0`. You will also need to rename the binary from `terraform-validator-<your-platform>` to `terraform-validator` and the terraform-validator binary must be in your PATH.
```
Comment (bharathkkb) [inline 0-bootstrap/README.md:111]: Let's link to the new TFV install docs

```suggestion
1. To run terraform-validator steps please follow the [instructions](https://github.com/GoogleCloudPlatform/terraform-validator/blob/main/docs/install.md) in the **Install Terraform Validator** section and install version `v0.4.0`. You will also need to rename the binary from `terraform-validator-<your-platform>` to `terraform-validator` and the terraform-validator binary must be in your PATH.
```

**Review by bharathkkb — APPROVED** [review]
LGTM with a minor suggestion

**Comment by vovinacci** [comment]
@googlebot I signed it!

**Comment by vovinacci** [comment]
@bharathkkb Updated.

**Pull request merged**